packages/components/modules/profiles/web/ProfileComponent/ReportButtonWithDialog/index.tsx

```diff
@@ -0,0 +1,231 @@
+'use client'
+
+import { FC, useMemo, useState } from 'react'
+
+import { Dialog } from '@baseapp-frontend/design-system/components/web/dialogs'
+import {
+  ChevronIcon,
+  FlagIcon,
+  OutlinedCheckMarkIcon,
+} from '@baseapp-frontend/design-system/components/web/icons'
+
+import { Box, Button, Divider, MenuItem, TextField, Typography } from '@mui/material'
+import { useLazyLoadQuery } from 'react-relay'
+
+import { ReportTypeListQuery as ReportTypeListQueryType } from '../../../../../__generated__/ReportTypeListQuery.graphql'
+import { useReportCreateMutation } from '../../../common/graphql/mutations/ReportCreate'
+import { ReportTypeListQuery } from '../../../common/graphql/queries/ReportTypeList'
+import { TypeButton } from './styled'
+import { ReportButtonWithDialogProps } from './types'
+
+type AllReportTypes = NonNullable<ReportTypeListQueryType['response']['allReportTypes']>
+type ReportType = NonNullable<AllReportTypes['edges'][number]>
+type ReportTypeNode = NonNullable<ReportType['node']>
+type ReportTypeSubType = NonNullable<ReportTypeNode['subTypes']['edges'][number]>
+type ReportTypeSubTypeNode = ReportTypeSubType['node']
+
+const ReportButtonWithDialog: FC<ReportButtonWithDialogProps> = ({ targetId, handleClose }) => {
+  const { allReportTypes } = useLazyLoadQuery<ReportTypeListQueryType>(ReportTypeListQuery, {
+    topLevelOnly: true,
+    targetObjectId: targetId,
+  })
+  const [isReportModalOpen, setIsReportModalOpen] = useState(false)
+  const [currentStep, setCurrentStep] = useState('report')
+  const [reportType, setReportType] = useState<ReportTypeNode>()
+  const [reportSubType, setReportSubType] = useState<ReportTypeSubTypeNode>()
+  const [reportText, setReportText] = useState('')
+
+  const [commitMutation, isMutationInFlight] = useReportCreateMutation()
+  const reportTypes = useMemo(
+    () => allReportTypes?.edges?.filter((edge) => edge?.node).map((edge) => edge?.node) || [],
+    [allReportTypes?.edges],
+  ).sort((a, b) => {
+    if (!a) return -1
+    if (!b) return 1
+    return a.key > b.key ? 1 : -1
+  })
+  const subTypes = useMemo(
+    () => reportType?.subTypes?.edges?.filter((edge) => edge?.node).map((edge) => edge?.node) || [],
+    [reportType?.subTypes?.edges],
+  )
+
+  const onClose = () => {
+    handleClose()
+    setCurrentStep('report')
```

Comment:
lets add constants to the variables on this file instead of using the strings

Reply:
Done